packages/language/src/preprocessor/pli-preprocessor-interpreter-state.ts

```
@@ -54,11 +58,18 @@ export class PliPreprocessorInterpreterState
     const newCounter = next(this.plainState.programCounter);
     if (newCounter >= 0 && newCounter < this.plainState.program.length) {
       this.plainState = { ...this.plainState, programCounter: newCounter };
+      const instruction = this.goToCounter.get(newCounter) ?? 0;
+      this.goToCounter.set(newCounter, instruction + 1);
+      // Ensure that we don't run into an infinite loop
+      if (instruction > MAX_GO_TO_COUNT) {
+        this._halt = true;
+        console.log("Long running preprocessor code detected. Stopping.");
```

_Question_: Would it make sense to surface this as a diagnostic instead of logging to the console? That way, users would get direct feedback if they accidentally introduce an infinite loop.